### PR TITLE
drm/meson: Fix scaler and enable underscan

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T15945

### CODE REVIEW

File: drivers/gpu/drm/meson/meson_drv.c [hunk 1]
```diff
@@ -1043,7 +1041,7 @@ static void update_scaler(struct drm_crtc *crtc)
 		/* Basic scaler config */
 		osd_reg_write(VPP_OSD_SC_CTRL0,
 			      (1 << 3) /* Enable scaler */ |
-			      (0 << 2) /* Select OSD1 */);
+			      (1 << 2) /* Select OSD1 */);
```
Comment (dsd) [inline drivers/gpu/drm/meson/meson_drv.c]: Can you explain this change in the commit message? i.e. did you check some docs or other code in order to understand how to configure VPP_OSD_SC_CTRL0?

Comment (magcius) [inline drivers/gpu/drm/meson/meson_drv.c]: I think the comment is incorrect -- probably my fault.

If you compare https://github.com/endlessm/linux-meson/blob/master/drivers/amlogic/display/osd/osd_hw.c#L1898 and https://github.com/endlessm/linux-meson/blob/master/drivers/amlogic/display/osd/osd_hw.c#L2093 you'll see that the OSD1 vs. OSD2 select is at bit position 0 (and the PDF agrees). Unfortunately, bit 2 doesn't seem to be labeled at all in the PDF I have, which seems to be the wrinkle.

Comment (carlocaione) [inline drivers/gpu/drm/meson/meson_drv.c]: I compared the initialization steps against the amlogic driver (and also considering how ORing with 0 doesn't make a lot of sense).

Comment (carlocaione) [inline drivers/gpu/drm/meson/meson_drv.c]: @dsd force-pushed with a new commit message.

Comment (magcius) [inline drivers/gpu/drm/meson/meson_drv.c]: Hm, the comment still isn't fixed though.

**Pull request merged**